operator/v1/types_network.go

```diff
@@ -454,25 +454,61 @@ type NetFlowConfig struct {
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=10
 	Collectors []IPPort `json:"collectors,omitempty"`
+
+	// flowCollectors defines the NetFlow collectors that will consume the flow data exported from OVS.
+	// It is a list of objects featuring a valid IP address and a port with a maximum of ten items
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	FlowCollectors []NetworkFlowTarget `json:"flowCollectors,omitempty"`
 }
 
 type SFlowConfig struct {
 	// sFlowCollectors is list of strings formatted as ip:port with a maximum of ten items
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=10
 	Collectors []IPPort `json:"collectors,omitempty"`
+
+	// flowCollectors defines the NetFlow collectors that will consume the flow data exported from OVS.
+	// It is a list of objects featuring a valid IP address and a port with a maximum of ten items
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	FlowCollectors []NetworkFlowTarget `json:"flowCollectors,omitempty"`
 }
 
 type IPFIXConfig struct {
 	// ipfixCollectors is list of strings formatted as ip:port with a maximum of ten items
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=10
 	Collectors []IPPort `json:"collectors,omitempty"`
+
+	// flowCollectors defines the NetFlow collectors that will consume the flow data exported from OVS.
+	// It is a list of objects featuring a valid IP address and a port with a maximum of ten items
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	FlowCollectors []NetworkFlowTarget `json:"flowCollectors,omitempty"`
 }
 
 // +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]):([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`
 type IPPort string
 
+type NetworkFlowTarget struct {
+	// ip address of the flow collector
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	IP string `json:"ip"`
+
+	// port number of the flow collector.
+	// If not specified, then the default port for each protocol will be used
+	//   - 2055 for NetFlow
```

Comment:
I can't say there's a default for netflow - but I saw in multiple places that this is the most common value.

But I still am not comfortable providing it as "default" in the code.

I would just make the port required, as it was before. 

Your thoughts @danwinship ?

Reply:
ping @danwinship can you think of an alternative ?